std-bits/base/src/main/java/org/enso/base/spi/EnsoServiceLoader.java

```diff
@@ -0,0 +1,65 @@
+package org.enso.base.spi;
+
+import java.util.List;
+import java.util.ServiceLoader;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import org.graalvm.polyglot.Value;
+
+/** A helper class that allows loading Enso services from the classpath. */
+public class EnsoServiceLoader<T extends EnsoService> {
+  private final ServiceLoader<T> loader;
+  private transient List<T> cachedProviders = null;
+
+  protected EnsoServiceLoader(Class<T> clazz) {
```

Comment:
There should be no need to subclass. Make the class `final`. Make the constructor `private`.